### PR TITLE
Modal component

### PR DESCRIPTION
**Pull request by mehulsatardekar**
Added Modal Component

Live Preview : https://framer-ui.netlify.app/docs/modal.html

### CODE REVIEW

File: components/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    border-radius: 10px;
+  }
+
+  .modal-width-80{
+    width: 80%;
+  }
+  
+  .model-content{
+     overflow: auto;
+     height: 330px;
```
Comment (Aazam10) [inline components/modal/modal.css]: You can use rems instead of pixel here

**Review by Aazam10 — COMMENTED** [review]
Instead of px try to use rem

File: components/modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+.modal {
+    position: fixed; 
+    z-index: 1;
+    left: 0;
+    top: 0;
+    width: 100%; 
+    height: 100%; 
+    overflow: auto; 
+    background-color: rgb(0,0,0); /* Fallback color */
+    background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
```
Comment (pritam-kr) [inline components/modal/modal.css]: you can use variables for colors.

File: components/modal/modal.css [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  }
+  
+  @media screen and (max-width:680px) {
+      
+      .modal-container{
+          margin: 20% auto;
+      }
+      
+  }
+
```
Comment (pritam-kr) [inline components/modal/modal.css:85]: remove these red mark by breaking line

**Review by pritam-kr — APPROVED** [review]
All looks good, Also your code looks neat and clean.

File: components/modal/modal.css [hunk 4]
```diff
@@ -0,0 +1,87 @@
+/* @import url('../../components/root.css');
+@import url('../../components/card/card.css');
+@import url('../../components/list/list.css'); */
+
```
Comment (KanekarSnehal) [inline components/modal/modal.css]: remove comments

File: components/modal/modal.css [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    width: 100%; 
+    height: 100%; 
+    overflow: auto; 
+    background-color: rgb(0,0,0); /* Fallback color */
+    background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
+    
+  }
+  
+  .modal-container {
+    background-color: #fefefe;
```
Comment (KanekarSnehal) [inline components/modal/modal.css]: use css variables for colors

File: components/modal/modal.css [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    margin: 5% auto;
+    padding: 10px;;
+    border: 1px solid #888;
```
Comment (KanekarSnehal) [inline components/modal/modal.css]: here also use css variables for colors

File: components/modal/modal.css [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+
+  
+  .model-container-bottom{
+    position: fixed;
+    bottom: 0;
+    background-color: #fefefe;
+    width: 100%;
+    height: 430px;
```
Comment (KanekarSnehal) [inline components/modal/modal.css]: use relative units like rem instead of absolute units

**Review by Dev-AkshatGupta — APPROVED** [review]
Everything is fine Just mentioned things that are mentioned by others need to be taken care of

File: components/modal/modal.css [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+     line-height: 2rem;
+     word-spacing: 1px;
+  }
+
+
+  
+  .model-container-bottom{
+    position: fixed;
+    bottom: 0;
+    background-color: #fefefe;
```
Comment (KanekarSnehal) [inline components/modal/modal.css]: use css variables

**Review by KanekarSnehal — COMMENTED** [review]
Good work, just make sure to remove comments and use relative units.

**Comment by mehulsatardekar** [comment]
@Aazam10 @KanekarSnehal made the changes as u suggested

File: docs/modal.html [hunk 9]
```diff
@@ -0,0 +1,428 @@
...
+                        <circle cx="16" cy="16" r="16" fill="white"></circle>
+                        <rect color="black" x="19" y="19" width="16" height="16" rx="8" ry="8"></rect>
+                    </mask>
+                    <foreignObject x="0" y="0" width="32" height="32" mask="url(#32e25648-9cb3-4aba-a168-64581e7aa254)">
+                        <div class="avatarStack-2Dr8S9"><img src="https://cdn.discordapp.com/avatars/709733291946672128/63f8ccaab30e072ae3a184ab142ddc54.webp" alt=" " class="avatar-user-icon" aria-hidden="true"></div>
+                    </foreignObject><svg x="14.5" y="17" width="25" height="15" viewBox="0 0 25 15">
+                        <mask id="2b3fa5a7-0ec2-4469-b164-f8d5007235b6">
+                            <rect x="7.5" y="5" width="10" height="10" rx="5" ry="5" fill="white"></rect>
+                            <rect x="12.5" y="10" width="0" height="0" rx="0" ry="0" fill="black"></rect>
+                            <polygon points="-2.16506,-2.5 2.16506,0 -2.16506,2.5" fill="black" transform="scale(0) translate(13.125 10)" style="transform-origin: 13.125px 10px;">
```
Comment (dev-prabhat) [inline docs/modal.html:146]: remove these inline style

File: components/modal/modal.css [hunk 10]
```diff
@@ -0,0 +1,85 @@
+
+@import url('../../components/framer.css');
+
+ :root{
+   --modal-bg-fallback :#rgb(0,0,0);
+   --modal-bg-fallback-op: rgba(0,0,0,0.4);
+   --modal-bg-container: #fefefe;
+   --modal-bg-border: #888;
```
Comment (dev-prabhat) [inline components/modal/modal.css:8]: try to consistent with same color code

**Review by dev-prabhat — CHANGES_REQUESTED** [review]
UI is clean, good work

**Review by dev-prabhat — APPROVED** [review]

**Review by Aazam10 — APPROVED** [review]

**Review by KanekarSnehal — APPROVED** [review]

**Pull request merged**